conan/tools/files/conandata.py

```diff
@@ -69,5 +69,8 @@ def trim_conandata(conanfile, raise_if_missing=True):
         if version_data is not None:
             result[k] = {version: version_data}
 
+    # Update the internal conanfile data too
```

Comment:
This is important @AbrilRBS @jcar87 
This was not done before, but it seems it make sense, otherwise ``self.conan_data`` in recipes can have outdated/trimmed information. Please check if this could be an issue in any ConanCenter flow.

Reply:
The possible issue would be if some custom command or something was reading ``conanfile.conandata`` after the ``export`` that trimmed it and somehow reading other versions information too. I doubt this happens ,but just in case

Reply:
mmm, we don't use `trim_conandata()` (yet) in any recipe in Conan Center, so I'm unsure as to how this change can be currently relevant? 

This makes sense regardless, but seems possibly unrelated to the PR?

Reply:
The result for the shake of validation should be the same, the important thing is to have the final exported ``conandata.yml`` trimmed or not. It doesn't matter if it is trimmed by the ConanCenter hook or by the recipe, what is important is that the ``export_conandata_patches`` is resilient to either case, having it trimmed or not, and that is what the new checks validate.